app/src/main/java/com/kylecombes/micrepeater/MainActivity2.java

```diff
@@ -37,6 +37,18 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        Boolean isFirstRun = getSharedPreferences("PREFERENCE", MODE_PRIVATE)
+                .getBoolean("isFirstRun", true);
+        //using a fake to run welcome wizard every time right now (for developing purposes). For app to function properly, replace fake with isFirstRun
+        Boolean fake = true;
+        if (fake) {
+            //show start activity
+            startActivity(new Intent(MainActivity2.this, WelcomeActivity.class));
+            //set isFirstRun to false so the welcome wizard doesn't run again on the same device
+            getSharedPreferences("PREFERENCE", MODE_PRIVATE).edit()
+                    .putBoolean("isFirstRun", false).commit();
```

Comment:
Replaced `commit()` with `apply()`, which will handle it in the background.

Reply:
what's the difference between commit and apply

Reply:
`apply()` will handle the saving in background. `commit()` will block the thread and let's say if we want to save a lot of data or the saving process takes longer than expected, there would be some lagging on the UI.